crypto/hash/hmac_ossl.c

```diff
@@ -95,146 +88,94 @@ static srtp_err_status_t srtp_hmac_alloc (srtp_auth_t **a, int key_len, int out_
     (*a)->out_len = out_len;
     (*a)->key_len = key_len;
     (*a)->prefix_len = 0;
-    new_hmac_ctx = (srtp_hmac_ctx_t*)((*a)->state);
-    memset(new_hmac_ctx, 0, sizeof(srtp_hmac_ctx_t));
+    new_hmac_ctx = (HMAC_CTX*)((*a)->state);
+    HMAC_CTX_init(new_hmac_ctx);
 
     return srtp_err_status_ok;
 }
 
 static srtp_err_status_t srtp_hmac_dealloc (srtp_auth_t *a)
 {
-    srtp_hmac_ctx_t *hmac_ctx;
+    HMAC_CTX *hmac_ctx;
 
-    hmac_ctx = (srtp_hmac_ctx_t*)a->state;
-    if (hmac_ctx->ctx_initialized) {
-        EVP_MD_CTX_cleanup(&hmac_ctx->ctx);
```

Comment:
In OpenSSL's FIPS User Guide https://www.openssl.org/docs/fips/UserGuide-2.0.pdf, the sample application using FIPS (page 136) uses HMAC_CTX.  So this should be fine.